src/kademlia/dht_storage.cpp

```diff
@@ -83,17 +84,15 @@ namespace
 	struct dht_immutable_item
 	{
 		// the actual value
-		std::unique_ptr<char[]> value;
+		std::vector<char> value;
```

Comment:
What do you think of reducing the size of `ips` to save that number of bytes? the use of vector is very convenient.